### PR TITLE
feat(data-exploration): split insight filters in the frontend

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

Our filters object has grown out of control

## Changes

On the frontend only, [splits the huge `FilterType` object](https://github.com/PostHog/posthog/compare/split-filter-type?expand=1#diff-19743365133d63884e16c2ed111a9076bb50a34785a3e624bd4abf49d49f814c) into multiple scoped filter objects like `TrendsFilterType`, `FunnelsFilterType`, etc.

## How did you test this code?

Lots of clicking, plus CI

### CODE REVIEW

**Comment by mariusandra** [comment]
Here's something small to review that could eliminate a class of bugs. 😅

File: frontend/src/lib/api.ts [hunk 1]
```diff
@@ -497,19 +497,19 @@ const api = {
             return await apiRequest.get()
         },
         async list(
-            filters: Partial<FilterType>,
+            filters: EventsListQueryParams,
```
Comment (pauldambra) [inline frontend/src/lib/api.ts:500]: I'd be tempted to call this `EventListFilter`. But nitting

**Review by pauldambra — APPROVED** [review]
There's an empty file but I know we're using it soon.

I'd be tempted to see what `FilterType` looks like as a Discriminated Union now but that's not blocking...

This is either a masterpiece of careful refactoring or has some fiendishly hidden bugs. 🥇

**Comment by mariusandra** [comment]
Thanks! I'll merge this in and fix the nits in a followup.

**Pull request merged**